packages/core/epoch/types.go

```diff
@@ -179,3 +176,128 @@ func (e *ECRecord) FromBytes(bytes []byte) (err error) {
 
 	return
 }
+
+type NodesActivityLog map[Index]*ActivityLog
```

Comment:
If these new types will still need to exist: is it necessary to place them as part of this package? Maybe they can be moved to the more specific components that use them.